src/zenml/models/v2/core/model.py

```diff
@@ -0,0 +1,314 @@
+    add_tags: Optional[List[str]]
+    remove_tags: Optional[List[str]]
+
+
+# ------------------ Response Model ------------------
+
+
+class ModelResponseBody(WorkspaceScopedResponseBody):
+    """Response body for models."""
+
+    tags: List["TagResponseModel"] = Field(
+        title="Tags associated with the model",
+    )
+    latest_version: Optional[str]
+    created: datetime = Field(
+        title="The timestamp when this component was created."
+    )
+    updated: datetime = Field(
+        title="The timestamp when this component was last updated.",
+    )
+
+
+class ModelResponseMetadata(WorkspaceScopedResponseMetadata):
+    """Response metadata for models."""
+
+    license: Optional[str] = Field(
+        title="The license model created under",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        title="The description of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    audience: Optional[str] = Field(
+        title="The target audience of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    use_cases: Optional[str] = Field(
+        title="The use cases of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    limitations: Optional[str] = Field(
+        title="The know limitations of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    trade_offs: Optional[str] = Field(
+        title="The trade offs of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    ethics: Optional[str] = Field(
+        title="The ethical implications of the model",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+
+
+class ModelResponse(
+    WorkspaceScopedResponse[ModelResponseBody, ModelResponseMetadata]
+):
+    """Response model for models."""
+
+    name: str = Field(
+        title="The name of the model",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+
+    def get_hydrated_version(self) -> "ModelResponse":
+        """Get the hydrated version of this model.
+
+        Returns:
+            an instance of the same entity with the metadata field attached.
+        """
+        from zenml.client import Client
+
+        return Client().zen_store.get_model(self.id)
+
+    # Helper methods
+    @property
+    def versions(self) -> List["ModelVersion"]:
+        """List all versions of the model.
+
+        Returns:
+            The list of all model version.
+        """
+        from zenml.client import Client
+
+        client = Client()
+        model_versions = client.list_model_versions(
+            model_name_or_id=self.id, page=1
+        )
+        ret = [
+            mv.to_model_version(suppress_class_validation_warnings=True)
+            for mv in model_versions.items
+        ]
+        for i in range(2, model_versions.total_pages + 1):
+            ret += [
+                mv.to_model_version(suppress_class_validation_warnings=True)
+                for mv in model_versions.items
+            ]
```

Comment:
This doesn't seem right, you iterate over `for mv in model_versions.items` N times here, I'm guessing the goal is to get all pages of model versions? If so, better use `pagination_utils.depaginate()` if you want to get ALL items